Connect 4 physics

ASecondGuy opened the pull request:
# Description
I added a new minigame.
It is a physics based version of connect 4.

## Type of change
- [X] New feature (non breaking change that adds functionality)

# Checklist
- [X] My code follows the general style guidelines of the project
- [x] I have perfomed a self-review of my code
- [x] I have commented my code, particulary where it is unclear
- [x] My changes generate no new warnings or errors
- [X] The project compiles and runs correctly

Numenter commented:
Node names should be in PascalCase. My reasoning is that godot names all standard nodes in PascalCase.

Some name suggestions.
- Connect/Bounds/SlopeLeft
- Connect/SpawnLeft

ASecondGuy commented:
@Numenter I ended up rewriting almost everything but you gave me a lot of inspiration.

Numenter commented on game/games/asecondguy_connect/connect.gd:
```diff
@@ -0,0 +1,118 @@
+
+# checks the chip and updates _end_condition
+func _check_game_end_from_chip(at: Vector2, player_id):
+	# check all 3 valid axis
+	for axis in [Vector2(1, 0), Vector2(0, 1), Vector2(1, 1)]:
```
```suggestion
	# check all 4 valid axis
	for axis in [Vector2(1, 0), Vector2(0, 1), Vector2(1, 1), Vector2(-1, 1)]:
```
![image](https://user-images.githubusercontent.com/84922125/174820114-d1491034-744e-4bd7-8812-1b47c26c812c.png)

ASecondGuy replied:
How did I forget this.

ASecondGuy committed:
add connect 4 physics game

lint fix

connect 4 ending working

chip detection working

gameplay working

fix codeorder and Nodenames

win condition check working

another win condition

end condition detection improved again

add some UI


re add stuff that got deleted somehow


fix lint


very basic (and stupid) connect ai 


connect player select


fix lint


fix lint again


fix lint again again


more player configuration


update icon


improve grid look with shader


fix lint


improved Ai


ai now considers the surounding chips


fix lint

Numenter commented:
Game speed is to slow. 
- It takes 4 sec to get the chip, that is way to long.
- The more FPS you got the slower moves the AI. (60fps = ca 4 sec, 400fps = ca. 32 sec. )[to lift the chip up]

Numenter commented:
Small thing that struck me.

It takes to long to get the chip (ca. 4 sec).

AI behaver
 - The AI moves the chip out of the frame. Yes the player can do that to, but I found that odd.
 - The AI should hover and drop the chip, the drop with momentum is odd.

The chip has to stop moving before the AI picks up the chip? I had one run where the left AI took longer to pick up the chips.

ASecondGuy commented:
I like the momentum thing because that is kinda in line with the physics part. 
I didn't think the other stuff was that bad but I guess I gotta fix that.

ASecondGuy committed:
speed up connect gameplay

ASecondGuy committed:
remove debug line
